logisland-engines/logisland-spark_1_6-engine/src/main/scala/com/hurence/logisland/stream/spark/KafkaRecordStreamParallelProcessing.scala

```diff
@@ -185,20 +188,17 @@ class KafkaRecordStreamParallelProcessing extends AbstractKafkaRecordStream {
                             outgoingEvents = processor.process(processorContext, incomingEvents)
 
                             /**
-                              * send metrics if requested
+                              * compute metrics
                               */
-                            processingMetrics.addAll(ProcessorMetrics.computeMetrics(
-                                appName,
-                                processorContext.getName,
-                                inputTopics,
-                                outputTopics,
-                                partitionId,
+                            val processorMetrics = ProcessorMetrics.computeMetrics(
+                                pipelineMetricPrefix + processorContext.getName + ".",
                                 incomingEvents,
                                 outgoingEvents,
                                 offsetRange.fromOffset,
                                 offsetRange.untilOffset,
-                                System.currentTimeMillis() - startTime))
+                                System.currentTimeMillis() - startTime)
 
+                            processorTimerContext.stop()
```

Comment:
Est-ce qu'il ne vaudrait pas lieux stopper le timer avant le computeMetrics ?

Reply:
can be a new issue